core/src/test/java/org/infinispan/xsite/AsyncBackupExpirationTest.java

```diff
@@ -0,0 +1,260 @@
+      XSiteAdminOperations adminOperations = extractComponent(cache(LON, 0), XSiteAdminOperations.class);
+      if (XSiteAdminOperations.OFFLINE.equals(adminOperations.siteStatus(NYC))) {
+         adminOperations.bringSiteOnline(NYC);
+      }
+
+      adminOperations = extractComponent(cache(NYC, 0), XSiteAdminOperations.class);
+      if (XSiteAdminOperations.OFFLINE.equals(adminOperations.siteStatus(LON))) {
+         adminOperations.bringSiteOnline(LON);
+      }
+   }
+
+   public AsyncBackupExpirationTest() {
+      super.lonBackupStrategy = BackupConfiguration.BackupStrategy.ASYNC;
+      super.nycBackupStrategy = BackupConfiguration.BackupStrategy.ASYNC;
+      super.implicitBackupCache = true;
+   }
+
+   @Override
+   protected ConfigurationBuilder getLonActiveConfig() {
+      return getConfig(lonConfigMode);
+   }
+
+   private AsyncBackupExpirationTest setLonConfigMode(ConfigMode configMode) {
+      this.lonConfigMode = configMode;
+      return this;
+   }
+
+   private AsyncBackupExpirationTest setNycConfigMode(ConfigMode configMode) {
+      this.nycConfigMode = configMode;
+      return this;
+   }
+
+   @DataProvider(name = "two boolean cross product")
+   public Object[][] tx() {
+      return new Object[][]{
+            {false, false},
+            {false, true},
+            {true, false},
+            {true, true}
+      };
+   }
+
+   private ControlledTimeService replaceTimeService() {
+      ControlledTimeService timeService = new ControlledTimeService();
+      // Max idle requires all caches to show it as expired to be removed.
+      for (Cache<?, ?> c : caches(LON)) {
+         TestingUtil.replaceComponent(c.getCacheManager(), TimeService.class, timeService, true);
+      }
+
+      for (Cache<?, ?> c : caches(NYC)) {
+         TestingUtil.replaceComponent(c.getCacheManager(), TimeService.class, timeService, true);
+      }
+
+      return timeService;
+   }
+
+   @Test(dataProvider = "two boolean cross product")
+   public void testExpiredAccess(boolean lifespan, boolean readOnPrimary) throws InterruptedException {
+      Cache<MagicKey, String> cache = cache(LON, 0);
+
+      ControlledTimeService timeService = replaceTimeService();
+
+      MagicKey key = readOnPrimary ? new MagicKey(cache) : new MagicKey(cache(LON, 1));
+      if (lifespan) {
+         cache.put(key, "v", 1, TimeUnit.SECONDS);
+      } else {
+         cache.put(key, "v", -1, TimeUnit.SECONDS, 1, TimeUnit.SECONDS);
+      }
+      eventuallyEquals("v", () -> cache(LON, 0).get(key));
+      eventuallyEquals("v", () -> cache(LON, 1).get(key));
+      Cache<Object, Object> backupCache = backup(LON);
+      assertNull(backupCache.get("k"));
+      eventuallyEquals("v", () -> backup(LON).get(key));
+
+      // Now expire the entry
+      timeService.advance(TimeUnit.SECONDS.toMillis(2));
+      assertNull(cache.get(key));
+
+      // TODO: check for the touch command being invoked on the remote site for lifespan = false
```

Comment:
actually, I would inspect the DataContainer to ensure the InternalCacheEntry's lastUsed is the expected value

Reply:
Yeah, that is what I meant by it actually.

Reply:
I moved it up a bit before the expiration to verify the last access was correct.